types.go

```diff
@@ -153,16 +153,15 @@ func (source *Source) MetadataWithAdditionalTags(tags []string) []MetadataField
 
 func (source *Source) AuthenticateToECR() bool {
 	logrus.Warnln("ECR integration is experimental and untested")
-	mySession := session.Must(session.NewSession(&aws.Config{
-		Region:      aws.String(source.AwsRegion),
-		Credentials: credentials.NewStaticCredentials(source.AwsAccessKeyId, source.AwsSecretAccessKey, source.AwsSessionToken),
-	}))
+	mySession := session.Must(session.NewSession())
 
 	var config aws.Config
 
 	// If a role arn has been supplied, then assume role and get a new session
 	if source.AwsRoleArn != "" {
-		config = aws.Config{Credentials: stscreds.NewCredentials(mySession, source.AwsRoleArn)}
+		config = aws.Config{Region: aws.String(source.AwsRegion), Credentials: stscreds.NewCredentials(mySession, source.AwsRoleArn)}
```

Comment:
Oops. I've approved based on our discussions. But I've just realised that this might break Crown because it automatically uses 'latest'. So when this thing rebuilds I don't think that'll be able to authenticate to ECR?

I think the if needs to be split in 2:

`if accesskeyid and accesskey are provided, set up a session with static credentials`
`if awsrolearn is provided then do a switch role`

That's beginning to look a lot like the previous code now, so can we go through this again on another call please?

Reply:
let's leave this here, and we can fix it up. AWS-Concourse is using this image but from my own repo.